amqp/src/main/java/io/github/tcdl/msb/adapters/amqp/AmqpMessageProcessingTask.java

```diff
@@ -34,10 +33,13 @@ public AmqpMessageProcessingTask(String consumerTag, String body, Channel channe
     public void run() {
         try {
             LOG.debug(String.format("[consumer tag: %s] Starting message processing: %s", consumerTag, body));
-            msgHandler.onMessage(body);
+            msgHandler.onMessage(body, ackHandler);
             LOG.debug(String.format("[consumer tag: %s] Message has been processed: %s", consumerTag, body));
+            ackHandler.autoConfirm();
```

Comment:
So we invoke `autoConfirm` (and hence fire AMQP ack) unconditionally after `onMessage` exit. Is it correct? What if the the developer wan't to fire an explicit ack in a separate thread, will it mean that with this call the ack will be sent prematurely?